creds/ssm/manager.go

```diff
@@ -0,0 +1,91 @@
+package ssm
+
+import (
+	"errors"
+	"io/ioutil"
+	"os"
+	"text/template"
+
+	"code.cloudfoundry.org/lager"
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/credentials"
+	"github.com/aws/aws-sdk-go/aws/session"
+	"github.com/concourse/atc/creds"
+)
+
+type SsmManager struct {
+	AwsAccessKeyID     string `long:"aws-access-key" description:"AWS Access key ID"`
+	AwsSecretAccessKey string `long:"aws-secret-key" description:"AWS Secret Access Key"`
+	AwsSessionToken    string `long:"aws-session-token" description:"AWS Session Token"`
+	AwsRegion          string `long:"aws-region" description:"AWS region to send requests to. Enviroment variable AWS_REGION is used if this flag is not provided."`
+	SecretTemplate     string `long:"secret-template" description:"AWS SSM parameter name template" default:"/{{.Team}}/{{.Pipeline}}/{{.Secret}}"`
```

Comment:
Regarding `"/{{.Team}}/{{.Pipeline}}/{{.Secret}}"`; as I commented in the issue, I was thinking that it might be nice to support a prefix (like `concourse/`), because without it we'd have to: 

- A: Give `/*` access to the ATC (not recommended).
- B: Change the policy every time you onboard a new team. 

However, in writing this comment I realise that this is not strictly needed for my use case (Concourse deployed using terraform) because I can just automate everything. But, perhaps it would still be nice to have for users that are doing e.g. Bosh deployments of Concourse?

Reply:
Since I haven't used parameter store before I did some sanity testing also, just throwing this in here in case it's useful in any way. Some rules:

- Parameter name cannot end with a `/`.
- Name cannot contain a `/` without also starting with a `/` (so `/some` and `/some/test` is allowed, `some/test` is not.)

I created 5 secrets:
- /some (just shows as 'some' in console)
- /some/test
- /some/test/value
- /other/secret

And tweaked the following policy:

```json
{
    "Version": "2012-10-17",
    "Statement": [
        {
            "Effect": "Allow",
            "Action": [
                "ssm:DescribeParameters"
            ],
            "Resource": "*"
        },
        {
            "Effect": "Allow",
            "Action": [
                "ssm:*"
            ],
            "Resource": "<resource>"
        },
        {
            "Effect": "Allow",
            "Action": [
                "kms:Decrypt"
            ],
            "Resource": [
                "arn:aws:kms:eu-west-1:<account>:key/*"
            ]
        }
    ]
}
```

And `<resource>` replaced as follows:

- `arn:aws:ssm:eu-west-1:<account>:parameter*` - Access to everything.
- `arn:aws:ssm:eu-west-1:<account>:parameter/*` - Access to everything.
- `arn:aws:ssm:eu-west-1:<account>:parameter/some*` - Access to all `/some*`
- `arn:aws:ssm:eu-west-1:<account>:parameter/some/*` - Access to all `/some/*`, but not `/some`.

#### My takeaway

It seems SSM Parameter store is meant to be used with paths. Scoping IAM policies based on path seems to work as expected. I.e., I think the approach taken in this PR make sense :)

Reply:
I have added `concourse-` as prefix in the template. Since the path can have [upto 5 levels](https://docs.aws.amazon.com/systems-manager/latest/APIReference/API_GetParametersByPath.html#EC2-GetParametersByPath-request-Path) only, it seemed prudent to provide more levels to the user. Policy setting should still work as expected since it does not care about the path.

Reply:
@surajbarkale 

I thought that if you reference a `((secret-value))` in a pipeline.yml, the current implementation of this (and other) secret backends will only check two paths:

- `/concourse/{{.Team}}/{{.Pipeline}}/secret-value`
- `/concourse/{{.Team}}/secret-value`

And that it's not possible to put e.g. `((nested/secret-value))` in pipeline.yml, and then retrieve `/concourse/{{.Team}}/{{.Pipeline}}/nested/secret-value`?

If I'm not mistaken about the above, I wonder if the users would actually be able to make use of the levels we save them by using `concourse-` as a prefix? (I.e., users would only run into trouble if they wanted to nest the concourse secrets under an existing path, which would require passing a new template via the command line anyway?)

Reply:
Ah, I initially thought that parameters allow `/` in names. Since that is not the case. I shall change to using 'concourse/' to match with existing [documentation](https://concourse.ci/creds.html#section_credential-lookup-rules).